src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -1317,15 +1317,38 @@ protected override bool SkipDependencyResolveError
 
             protected override IPackage ResolveDependency(NuGet.PackageDependency dependency)
             {
-                return _repository.ResolveDependency(dependency, allowPrereleaseVersions: false, preferListedPackages: false);
+                return _repository.ResolveDependency(dependency, allowPrereleaseVersions: false, preferListedPackages: true);
             }
 
             protected override bool OnAfterResolveDependency(IPackage package, IPackage dependency)
             {
-                _packages.Remove(dependency);
+                if (!FindDependency(dependency.Id, dependency.Version))
+                {
+                    _packages.Remove(dependency);
+                }
+
                 return base.OnAfterResolveDependency(package, dependency);
             }
 
+            private bool FindDependency(string id, SemanticVersion version)
+            {
+                foreach (var package in _packages)
+                {
+                    foreach (var set in package.DependencySets)
+                    {
+                        foreach (var dependency in set.Dependencies)
+                        {
+                            if (dependency.Id == id && dependency.VersionSpec.MinVersion < version)
```

Comment:
The version check should use Satisfies, this isn't taking into account ``>=`` or if the min version is included in the range